eventtracking/helpers.py

```diff
@@ -0,0 +1,49 @@
+"""
+Helper methods and classes for eventtracking
+"""
+import json
+from datetime import date, datetime
+from pytz import UTC
+
+from eventtracking.backends.caliper import CaliperBackend
+from eventtracking.backends.xapi import XAPIBackend
+
+
+class GenericJSONEncoder(json.JSONEncoder):
+    """
+    Custom JSON encoder that encodes asynchronous event backends and datetime objects to json.
+    """
+    def default(self, obj):  # pylint: disable=method-hidden
+        if isinstance(obj, CaliperBackend):
+            return {"type": "CaliperBackend", "dict": obj.__dict__}
```

Comment:
We are going to use processors for transforming, filtering and (possibly but not sure at the moment) routing purposes.
Those processors are initialized at the creation time of the tracker. Here, we are only decoding/encoding the constructor parameters. That means processor classes will not be initialized automatically.
We will have to manually import modules using `importlib`.

Reply:
We are most like to pass processors/filters like this then it should work.
`xapi': {
                    'ENGINE': 'eventtracking.backends.xapi.XAPIBackend',
                    'OPTIONS': {
                        'filters': ['path.to.filter.module', 'path.to.filter.module2'] 
                    }
                }`

Reply:
Exactly. We will be passing these module paths as strings and in our backend we would have to import these ourselves. While if we use the `processors` key, the tracker backend will take care of it and import and initialize it itself saving us from the trouble.

Just one caveat is that we will he to use the structure
```
`OPTIONS':[
    {
       'ENGINE':'path.to.module',
        'OPTIONS':{}
    }
]
```

Reply:
No. We don't need to nest `OPTIONS`. Only one level of `OPTIONS` should be enough passing different modules in them.